src/ed25519.rs

```diff
@@ -490,7 +491,7 @@ impl KeyPair for Ed25519KeyPair {
     }
 
     fn private(self) -> Self::PrivKey {
-        self.secret
+        Ed25519PrivateKey::from_bytes(self.secret.as_ref()).unwrap()
```

Comment:
Is the idea of this being we drop ‘self’ and zeroize it vs risking the compiler copying the secret in memory and leaving it there?

Reply:
I think this is brought by implementing `Drop` which prevents you from moving out of self. I have my issues with this `private` method ...